TAKAware/Screens/MainScreens/MapView.swift

```diff
@@ -508,6 +536,97 @@ struct MapView: UIViewRepresentable {
         updateAnnotations(from: view)
     }
     
+    private func kmlChangeNotified(notification: Notification) {
+        TAKLogger.debug("[MapView] Notified of a KML change \(notification.debugDescription)")
+        updateKmlOverlays()
+    }
+    
+    private func updateKmlOverlays() {
+        var processedKmlFiles: [String] = []
+        
+        dataContext.perform {
+            let fetchKml: NSFetchRequest<KMLFile> = KMLFile.fetchRequest()
+            fetchKml.predicate = NSPredicate(format: "visible = YES")
+            let fetchedKml = try? dataContext.fetch(fetchKml)
+            let incomingKml = (fetchedKml == nil) ? [] : fetchedKml!
+
+            incomingKml.forEach { kmlRecord in
+                guard let filePath = kmlRecord.filePath,
+                        let fileId = kmlRecord.id else {
+                    TAKLogger.debug("[MapView] KMLRecord had no name or file path")
+                    return
+                }
+                processedKmlFiles.append(fileId.uuidString)
+                
+                if loadedKmlAnnotations.contains(fileId.uuidString) {
+                    let existingAnnotations: [MapPointAnnotation] = mapView.annotations.filter {
+                        ($0 as? MapPointAnnotation)?.groupID == fileId
+                    } as! [MapPointAnnotation]
+                    TAKLogger.debug("[MapView] Found \(existingAnnotations.count) existing annotations for this KML")
+                    if !kmlRecord.visible {
```

Comment:
Need to make sure this works for KMZs as well. It should since we're switching on the groupID, but need to confirm.

Reply:
Confirmed visibility change works for KMZ and KMLs